lib/cocoapods/target/framework.rb

```diff
@@ -0,0 +1,30 @@
+module Pod
+  class Framework
+    # @return [String] the path to the .framework
+    #
+    attr_reader :source_path
+
+    # @return [String, Nil] the dSYM path, if one exists
+    #
+    attr_reader :dsym_path
+
+    def initialize(source_path, dsym_path)
+      @source_path = source_path
+      @dsym_path = dsym_path
+    end
+
+    def ==(other)
+      if other.class == self.class
+        other.source_path == @source_path && other.dsym_path == @dsym_path
+      else
+        false
+      end
+    end
+
+    alias eql? ==
+
+    def hash
+      source_path.hash
```

Comment:
was this generated?

Reply:
nope - did I do this incorrectly? I'm mostly used to having to implement equality in hashing for custom models but I'm still a noob at Ruby